spec/chartmogul/api_resource_spec.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'pry'
+
+describe ChartMogul::APIResource do
+  describe 'connection', vcr: true do
+    it 'works when credentials are updated' do
+      set_invalid_credentials
+      expect { ChartMogul::Ping.ping }.to raise_error(ChartMogul::UnauthorizedError)
+
+      set_valid_credentials
+      expect { ChartMogul::Ping.ping }.not_to raise_error
+    end
+
+    it 'works in a threaded environment' do
+      set_invalid_credentials
+      expect { ChartMogul::Ping.ping }.to raise_error(ChartMogul::UnauthorizedError)
+
+      Thread.new do
+        set_valid_credentials
+        expect { ChartMogul::Ping.ping }.not_to raise_error
+      end.join
```

Comment:
and now add another `expect { ChartMogul::Ping.ping }.to raise_error(ChartMogul::UnauthorizedError)` in the main thread without setting again the credentials, which will confirm that the connection was indeed invalidated.